### PR TITLE
added add/remove to cart functionality

### PR DESCRIPTION
**Pull request by vidhi-thakur**
- added context for cart
- updated values in navbar, cart-management, and product page
- fetching products from api

check preview URL [here](https://deploy-preview-5--savvy-react-store.netlify.app/)

### CODE REVIEW

File: src/reducers/cartReducer.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const initialCart = {
+    cartItems: [],
+    totalAmount: 0
+}
+
+function cartReducer(state, action) {
```
Comment (sonishreyas) [inline src/reducers/cartReducer.js]: you can destructure action here into type and payload

**Review by sonishreyas — APPROVED** [review]
The code looks good and easy to understand. I had 1 suggestion which you can implement. Rest looks good

File: src/components/cart-management/cart-management-card/CartManagementCard.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
-import React from 'react'
+import React from 'react';
+// context
+import { useCart } from 'context/addToCart';
 
-function CartManagementCard({ title, author, rating, description, price, cartImage, }) {
+function CartManagementCard({ id, title, author, rating, description, price, productImage, }) {
+    const [, dispatch] = useCart();
```
Comment (NeerajVodala) [inline src/components/cart-management/cart-management-card/CartManagementCard.js:6]: can use object here instead of array

**Review by NeerajVodala — APPROVED** [review]
app is breaking on refresh. Add redirects to public folder.
Rest everything looks good.

**Comment by vidhi-thakur** [comment]
Thank you @NeerajVodala for review and reporting the issue, will definitely fix  the bug.
Thank you @sonishreyas  for your suggestion

will merge the code after making the changes

**Pull request merged**